[Docs] update agent instructions to clarify package build commands (#36995)
Azure/azure-sdk-for-js: AGENTS.md

## Build, Test, and Development Commands
- `pnpm install` — install workspace deps (Node current LTS, pnpm v10).

> - **Building packages**: Due to workspace linking, `npm run clean && npm run build` under a package directory may fail if dependencies aren't built. Always use: `pnpm turbo build --filter={package name}... --token 1` (the trailing `...` builds the package and all its dependencies; `--token 1` enables remote cache read).

Author: ```suggestion
- **Building packages**: Due to workspace linking, `npm run clean && npm run build` under a package directory will NOT work if dependencies aren't built. Always use: `pnpm turbo build --filter=<package name>... --token 1` (the trailing `...` builds the package and all its dependencies; `--token 1` enables remote cache read).
```
